### PR TITLE
Fix person and organization filter for affiliations

### PR DESCRIPTION
**Pull request by lsetiawan**
## Overview

This PR addresses #95. It joins the necessary table to `Affiliations` table so that filter can actually occur. I have tested it and it seems to work.

### CODE REVIEW

**Comment by emiliom** [comment]
So, a `join` is necessary, and you can't use the [relationships defined](https://github.com/ODM2/ODM2PythonAPI/blob/master/odm2api/ODM2/models.py#L192) in the `Affiliations` model? eg, `OrganizationObj = relationship(Organizations)`

**Comment by lsetiawan** [comment]
@emiliom How do I go about that? Sorry.. not SQLAlchemy pro here 😉

**Comment by emiliom** [comment]
> @emiliom How do I go about that? Sorry.. not SQLAlchemy pro here

Neither am I! But I'd look for similar examples in other query functions.

If it is in fact doable, it seems cleaner to reuse that "predefined" relationship rather than adding explicit joins.

File: odm2api/ODM2/services/readService.py [hunk 1]
```diff
@@ -554,11 +554,12 @@ def getAffiliations(self, ids=None, personfirst=None, personlast=None, orgcode=N
         q = self._session.query(Affiliations)
 
         if ids: q = q.filter(Affiliations.AffiliationID.in_(ids))
-        if orgcode: q = q.join(Organizations).filter(Organizations.OrganizationCode.ilike(orgcode))
-        if personfirst or personlast:
-            q = q.join(People)
-            if personfirst: q = q.filter(People.PersonFirstName.ilike(personfirst))
-            if personlast: q = q.filter(People.PersonLastName.ilike(personlast))
+        if orgcode: q = q.join(Affiliations.OrganizationObj).filter(
```
Comment (lsetiawan) [inline odm2api/ODM2/services/readService.py]: @emiliom Is this what you're looking for? I still have to do the join... Can't find anything that simply filters straight up from relationship.

Comment (lsetiawan) [reply odm2api/ODM2/services/readService.py]: I got the sample from https://github.com/ODM2/ODM2PythonAPI/blob/master/odm2api/ODM2/services/readService.py#L421

**Comment by emiliom** [comment]
Hmm. That doesn't look better :disappointed: 

After looking at what's used in `readService.py`, I can see that there's some variability, and the two join forms you've used (in the two commits) are what's used. Oh well.

If you've tested this, let's go with it. It would seem cleaner and *possibly* more efficient (in some cases) to do all the joining in the `query` statement, but let's not bother.

Thanks.

**Comment by emiliom** [comment]
Let me know if you've tested this commit, and I'll accept the PR.

**Comment by lsetiawan** [comment]
This has been tested and good to go with your approval. Thanks!

**Pull request merged**